### DISCUSSION TITLE
FABM-DEB know how

### DISCUSSION THREAD

**Discussion by jpein1** [fabm-model/fabm#51] [Q&A]
Hi, does anybody have experience in running the DEB model with FABM? My first question would be how the "line structure" can be integrated into a modelling framework HYDROCORE-FABM-BGCMODEL (e.g. SCHISM-FABM-ECOSMO). Second question would be how can i compile two models like ECOSMO and DEB together and make them exchange information via FABM? 
Best regards,
Johannes

**Comment by jornbr** [comment]
Hi Johannes,

There's [an existing DEB-based structured population model for FABM](https://github.com/jornbr/fabm-deb) - is that the one you are considering using? Or are you planning a new implementation of a DEB model? In any case, the existing one is a 2D (depth-integrated) model written to consume a depth-integrated food source. It has been reasoanbly well tested, but it is poorly documented. It would need a little work to be able to interact with a pelagic model such as ECOSMO, though some [recently developed FABM functionality](https://doi.org/10.5281/zenodo.10600259) makes such 2D-3D mapping easier.

Further details will depend on what you need exactly. For instance:

- Do you need a model for an individual (or nearly equivalent: a population where all individuals at any given location are 100% identical, as you might assume for e.g., mussel farms), or a structured population model that allows for overlapping generations?
- Where in the water column will your DEB organisms be? (does "line structure" refer to mussel lines in aquaculture?) Should the DEB organisms be transported along with the water? If not, do you want to resolve variability in the population in the vertical, or are you happy to assume the DEB population is the same throughout the water column? (the former means you need a depth-explicit model, the latter means a depth-integrated model could suffice).

Cheers,

Jorn

**Reply by jpein1** [reply]
Hi Jorn,

thanks, I found the existing DEB model that you linked above and I consider to use it to later modify it towards a specific organism. 
For the moment I'd be fine with a depth-integrated model. 
Yes the model should represent a population of identical organisms - infact the idea is to simulate a mussel farm. 
I am not sure where the mussel lines will be placed vertically in the end, but the water is shallow and 2D should be perfectly fine. The organisms shall stay in place and interact with the biogeochemical environment (e.g. ECOSMO).

Hopefully it is not a problem of the hydro-bgc model is 3D and the DEB is 2D? 

I now work with an older FABM version (from 2019/2020) and I thought I could just copy the DEB to my working souce code. However I found that several routines such as _ADD_BOTTOM_SOURCE_ do not work with the version I use. What would you suggest how to move forword? 

Best,
Johannes

**Reply by jornbr** [reply]
Hi Johannes,

In principle, if you need a population of identical organisms, the existing DEB code is overkill - it does a lot of effort to simultaneosuly track multiple cohorts in the population, and that adds complexity that you could conceivably do without. Still, it might provide a way to get started quickly.

It should be fine to use the 2D DEB model in a 3D host - in fact, it was originally developed and tested for use with (3D) FVCOM. However,  as you've seen, it does require a recent version of FABM: v1 or above. If you are still on lower version (the lack of `_ADD_BOTTOM_SOURCE_` suggests that this is the case), I'd recommend [upgrading first](https://github.com/fabm-model/fabm/wiki/FABM-1.0). If that requires updating the host coupling layer, it will require some effort. That will pay off though - it is likely to improve performance a little, and also gives you access to a much wider range of FABM models and building blocks, including the 2D-3D mapping logic that will come in useful when linking DEB with ECOSMO.

Cheers,

Jorn
